Dockerfile

```diff
@@ -16,57 +6,60 @@ FROM ghcr.io/graalvm/graalvm-ce:ol8-java17-22 AS graalvm-native-image-jdk17
 
 COPY --from=azul/zulu-openjdk:7 /usr/lib/jvm/zulu7 /usr/lib/jvm/zulu7
 COPY --from=azul/zulu-openjdk:8 /usr/lib/jvm/zulu8 /usr/lib/jvm/zulu8
 COPY --from=azul/zulu-openjdk:11 /usr/lib/jvm/zulu11 /usr/lib/jvm/zulu11
-COPY --from=azul/zulu-openjdk:17 /usr/lib/jvm/zulu17 /usr/lib/jvm/zulu17
 
 COPY --from=ibmjava:8-sdk /opt/ibm/java /usr/lib/jvm/ibm8
-COPY --from=ibm-semeru-runtimes:open-11-jdk-jammy /opt/java/openjdk /usr/lib/jvm/ibm11
-COPY --from=ibm-semeru-runtimes:open-17-jdk-jammy /opt/java/openjdk /usr/lib/jvm/ibm17
 
 COPY --from=ibm-semeru-runtimes:open-8-jdk-jammy /opt/java/openjdk /usr/lib/jvm/semeru8
+COPY --from=ibm-semeru-runtimes:open-11-jdk-jammy /opt/java/openjdk /usr/lib/jvm/semeru11
+COPY --from=ibm-semeru-runtimes:open-17-jdk-jammy /opt/java/openjdk /usr/lib/jvm/semeru17
 
 COPY --from=graalvm-native-image-jdk11 /opt/graalvm-ce-java11-22* /usr/lib/jvm/graalvm22-jdk11
 COPY --from=graalvm-native-image-jdk17 /opt/graalvm-ce-java17-22* /usr/lib/jvm/graalvm22-jdk17
 
-COPY --from=oracle8 /usr/lib/jvm/oracle8 /usr/lib/jvm/oracle8
+RUN sudo apt-get -y update && sudo apt-get -y install curl
+# See: https://gist.github.com/wavezhang/ba8425f24a968ec9b2a8619d7c2d86a6
+RUN set -eux; \
+    sudo mkdir -p /usr/lib/jvm/oracle8; \
+    curl -L --fail "https://javadl.oracle.com/webapps/download/AutoDL?BundleId=246284_165374ff4ea84ef0bbd821706e29b123" | sudo tar -xvzf - -C /usr/lib/jvm/oracle8 --strip-components 1
 
-RUN sudo apt-get -y clean && sudo rm -rf /var/lib/apt/lists/*
+RUN sudo rm -rf \
+    /usr/lib/jvm/*/man \
+    /usr/lib/jvm/*/lib/src.zip \
+    /usr/lib/jvm/*/demo \
+    /usr/lib/jvm/*/sample \
+    /usr/lib/jvm/graalvm*/lib/installer
 
-# Install some common useful things
-RUN set -eux; \
-    sudo apt-get update; \
-    sudo apt-get dist-upgrade; \
-    sudo apt-get install apt-transport-https socat; \
-    sudo apt-get install vim less debian-goodies; \
-    sudo apt-get install autossh;
+FROM cimg/base:edge-22.04
 
-# Install aws cli
-RUN set -eux; \
-    sudo apt install python3-pip; \
-    pip3 install awscli;
+COPY --from=builder /usr/lib/jvm /usr/lib/jvm
 
-# Install autoforward script
 COPY autoforward.py /usr/local/bin/autoforward
-RUN set -eux; \
-    sudo pip3 install requests requests-unixsocket; \
-    sudo chmod +x /usr/local/bin/autoforward
-
-# Install datadog-ci
-RUN sudo curl -L --fail "https://github.com/DataDog/datadog-ci/releases/download/v1.3.0-alpha/datadog-ci_linux-x64" --output "/usr/local/bin/datadog-ci" \
-    && sudo chmod +x /usr/local/bin/datadog-ci
 
-RUN sudo rm -rf /tmp/..?* /tmp/.[!.]* /tmp/*
+RUN set -eux; \
+    sudo apt-get update; \
+    sudo apt-get install --no-install-recommends apt-transport-https socat; \
+    sudo apt-get install --no-install-recommends vim less debian-goodies; \
+    sudo apt-get install --no-install-recommends autossh; \
+    sudo apt install python3-pip; \
+    sudo apt-get -y clean; \
```

Comment:
Adding `apt-get autoremove --yes` might be a nice addition here (not sure if it will remove a lot of non needed packages though)

Reply:
AFAIK removing files that are already present in a previous layer does not really improve the size of the download (if anything, it'll make it worse). The only way to reduce the size is avoiding the installs in the first place.